tests/core/fs/workspacefs/test_file_history.py

```diff
@@ -0,0 +1,65 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import pytest
+
+from parsec.core.fs.workspacefs.versioning_helpers import VersionLister
+from parsec.core.types import FsPath
+
+
+@pytest.mark.trio
+async def test_file_history(alice_workspace):
+    sync_by_id = alice_workspace.sync_by_id
+    entry = alice_workspace.get_workspace_entry()
+    wid = entry.id
+
+    # Empty workspace
+    await sync_by_id(wid)
+
+    # Creating files and writing first byte
+    await alice_workspace.touch("/f")
+
+    # Test version before first sync
+    f_versions = await VersionLister(alice_workspace).list(FsPath("/f"))
+
+    # Assert if task list have been completed
+    assert f_versions[1] is True
+
+    # Should have no version yet because we didn't sync
+    assert not f_versions[0]
+
+    # Syncronysing the files
+    await sync_by_id(wid)
+
+    # Checking again, should have the first version now
+    f_versions = await VersionLister(alice_workspace).list(FsPath("/f"))
+
+    # Assert if task list have been completed
+    assert f_versions[1] is True
+    # Assert there is a version in the list
+    assert f_versions[0]
+    # Checking the first version of the list
+    assert getattr(f_versions[0][0], "version") == 1
```

Comment:
Hard to read statement, it would be much simpler to write:
```python
assert versions_list[0].version == 1
```
or even:
```python
assert versions_list == [
    TimestampBoundedData(
        id=ANY,
        version=1,
        early=ANY,
        late=ANY,
        creator=alice.device_id,
        updated=ANY,
        is_folder=False,
        size=0,
        source=ANY,  # not sure what to put here...
        destination=ANY,  # not sure what to put here...
    )
]
```
This has the added benefit of checking the number of items in `versions_list`
The check against a `TimestampBoundedData` is a bit cumbersome, so it really depend on many field inside it we want to test and how complex to create this object is...